src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -113,6 +114,52 @@ public static long BigMul(int a, int b)
             return ((long)a) * b;
         }
 
+        [CLSCompliant(false)]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static ulong BigMul(ulong a, ulong b, out ulong low)
+        {
+            if (Bmi2.X64.IsSupported)
+            {
+                ulong tmp, high;
+                unsafe
```

Comment:
I'd, personally, rather just mark the entire method unsafe than have this `unsafe` block in the middle of the code.